rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3188,6 +3127,37 @@ extern "C" rmw_ret_t rmw_get_node_names_with_security_contexts(
   }
   return RMW_RET_BAD_ALLOC;
 }
+
+extern "C" rmw_ret_t rmw_get_node_names(
+  const rmw_node_t * node,
+  rcutils_string_array_t * node_names,
+  rcutils_string_array_t * node_namespaces)
+{
+  return rmw_get_node_names_impl(
+    std::regex("^name=([^;]*);namespace=([^;]*);securitycontext=([^;]*);", std::regex::extended),
+    node,
+    node_names,
+    node_namespaces,
+    nullptr);
+}
+
+#if RMW_VERSION_GTE(0, 8, 2)
+extern "C" rmw_ret_t rmw_get_node_names_with_security_contexts(
+  const rmw_node_t * node,
+  rcutils_string_array_t * node_names,
+  rcutils_string_array_t * node_namespaces,
+  rcutils_string_array_t * security_contexts)
+{
+  if (RMW_RET_OK != rmw_check_zero_rmw_string_array(security_contexts)) {
+    return RMW_RET_ERROR;
+  }
```

Comment:
@ivanpauno is it invalid to call `rmw_get_node_names_with_security_contexts()` with a null `security_contexts`? If not, I think this check should be relocated to the implementation function and conditionally executed based on `security_contexts` value.

Reply:
In `rmw_get_node_names_impl` `security_contexts` can be `NULL`.
In `rmw_get_node_names_with_security_contexts` `security_contexts` cannot be `NULL`.

I think the current logic is fine.